### ISSUE TITLE
Import not working with webpack+babel

### ISSUE THREAD

**Issue by dlwalsh** [chartist-js/chartist#403]
I'd like to use chartist in the manner below. My workflow uses webpack and babel.

> import Chartist from 'chartist';

This throws the following in the browser: `Uncaught TypeError: Cannot set property 'Chartist' of undefined`

I have got around this by forking chartist and tweaking the factory code in the manner described here: https://github.com/gregjacobs/Autolinker.js/issues/86#issuecomment-74487812

It would be great if this could be incorporated into the master branch.

**Comment by gionkunz** [comment]
hey there. Thanks for the report. We use grunt-umd which is using the UMD pattern from Adi Osmani. Can you check against grunt-umd? I know they are using templates that can also be customized. Not sure if we should just override those. Ill also take a look whats haplening never used it with webpack but browserify and systemjs.

**Comment by DavidWells** [comment]
+1 on this. I fixed the same way with https://github.com/gregjacobs/Autolinker.js/issues/86#issuecomment-74487812

> if I changed https://github.com/gregjacobs/Autolinker.js/blob/master/dist/Autolinker.js#L2
> to if (root && typeof define === 'function' && define.amd) {

**Comment by hambyiii** [comment]
+1 from me as well. Would rather use Chartist than D3 but lack of webpack+babel support will push me off.

**Comment by kinetifex** [comment]
+1 Any updates or timeline on this?

**Comment by opensourceportfolio** [comment]
was anyone able to find a clean solution without resorting to forking the code?

**Comment by chiefcll** [comment]
https://github.com/gionkunz/chartist-js/pull/772

**Comment by gionkunz** [comment]
Thanks @chiefcll . This will be distributed with the next release.

**Comment by maxknee** [comment]
When will the new release go out?

**Comment by gionkunz** [comment]
0.9.9 is scheduled for the upcoming weekend

**Comment by ricardobanegas** [comment]
For me this is still broken with latest version `0.10.1`. 

`Uncaught TypeError: Cannot set property 'Chartist' of undefined`

When using latest Webpack 2 + Babel + [babel-preset-es2015](https://github.com/babel/babel/tree/master/packages/babel-preset-es2015)

`import Chartist from 'chartist'`

Update: Problem gets resolved when updating to babel-preset-es2016, but this is not ideal. 

Any ideas on how to solve this?

**Comment by ricardobanegas** [comment]
Actually, got resolved when bumping up to babel-preset-es2016. Just in case anyone else gets this problem.

**Comment by gionkunz** [comment]
Maybe someone on this thread can answer? I don't use Chartist on webpack that much.

**Comment by ricardobanegas** [comment]
I still have the error, so I had to patch it myself. I guess the problem lies within the way Chartist declares the module for universal use. 

Maybe this article shares some light on the subject? 
[Javascript Library starter using Webpack & ES6](http://krasimirtsonev.com/blog/article/javascript-library-starter-using-webpack-es6)

Also, I noticed that you're using grunt-umd, and it has been patched to 2.4.0, could be worth to update and see if that solves the problem.

**Comment by gionkunz** [comment]
Ok, I'll create a task to update to the latest grunt-umd.

**Comment by noorhammad** [comment]
Does anyone have a patch for the webpack issue?

https://github.com/gregjacobs/Autolinker.js/issues/86#issuecomment-74487812

Does not resolve.

**Comment by sternert** [comment]
@ricardobanegas "so I had to patch it myself":
How did you patch it?

**Comment by powerbuoy** [comment]
Any progress on this? I'm having the same issue with Aurelia CLI.

**Comment by MartinMuzatko** [comment]
Import works flawlessly. As stated in another issue,  not all plugins return functions.

`import Chartist from 'chartist'` worked perfectly.

**Comment by MartinMuzatko** [comment]
Now having the same problem with `babel-preset-env` ^^'

**Comment by hovancik** [comment]
I'm trying to use Chartists with Rails 5.2 (webpacker) and import does not work for me.

**Comment by wiscow** [comment]
having this issue also. Importing 'chartist' does not seem to resolve the issue.  the Js resolves but having issues with the css.

**Comment by mlg87** [comment]
three years later, and this still seems to be an issue. why is this closed? both `import Chartist from 'chartist'` and `require('chartist')` lead to this error for me

edit: using `0.10.1`

**Comment by math0ne** [comment]
I fixed this by changing:

```
(function (root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module unless amdModuleId is set
    define('Chartist', [], function () {
      return (root['Chartist'] = factory());
    });
  } else if (typeof module === 'object' && module.exports) {
    // Node. Does not work with strict CommonJS, but
    // only CommonJS-like environments that support module.exports,
    // like Node.
    module.exports = factory();
  } else {
    root['Chartist'] = factory();
  }
}(this, function () {
```

to this:

```
(function (root, factory) {
  if (typeof module === 'object' && module.exports) {
    // Node. Does not work with strict CommonJS, but
    // only CommonJS-like environments that support module.exports,
    // like Node.
    module.exports = factory();
  } else {
    root['Chartist'] = factory();
  }
}(this, function () {
```

I'm sure I'm doing something wrong but this worked for me.

**Comment by besworks** [comment]
This same problem exists when trying to include Chartist inside an ES Module. I tracked the issue down to two problems... First, the `global` object doesn't exist so I added `if (!window.global) { window.global = window; }` at the top of the file. Then, on the last line of the code block that math0ne posted above me, I changed `(this, function () {` to `(this || global, function () {`. Or, in the minified version: `a.Chartist=b()}(this,function(){var a={version:"0.11.4"};` would become `a.Chartist=b()}(this||global,function(){var a={version:"0.11.4"};`. Then just import the entire script with `import './chartist.min.js';` which sets up the global `Chartist` object as expected.
